src/internal/consistenthashing/etcd.go

```diff
@@ -0,0 +1,370 @@
+package consistenthashing
+
+import (
+	"context"
+	"fmt"
+	"hash/crc32"
+	"path"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/v2/src/internal/collection"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/uuid"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/dlock"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/watch"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/sirupsen/logrus"
+	etcd "go.etcd.io/etcd/client/v3"
+)
+
+const (
+	// LeaseTTL is used by nodes in a ring to refresh their liveness state.
+	LeaseTTL = int64(5) // number of seconds between lease renewals.
+)
+
+// Ring is a consistent hash Ring. Each process should only create one instance with a given prefix. Nodes can be
+// added locally to a Ring. A Ring watches for changes to the prefix to determine if new nodes have been added
+// or deleted.
+// A Ring should be initialized with a call to New.
+type Ring struct {
+	Client *etcd.Client   // Client will be used to make api calls against etcd.
```

Comment:
The field comments seem like overkill, and add some amount of maintenance overhead. I think non-trivial functionality / function comments make sense, but fields / variables should normally be self documenting based on there name and only needing comments in exceptional cases.